pkg/volume/azure_file/azure_file.go

```diff
@@ -0,0 +1,236 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure_file
+
```

Comment:
@saad-ali Provisioning Azure is not as straightforward as the other 3.   The one's we have now have a cloud provider and an API and they all work the same way.  Azure requires different authn/authz, needs user secrets, etc.

I think designing and implementing all that is out of scope for 1.2.  We'll need to design the next version of provisioning with this stuff in mind.